### DISCUSSION TITLE
4.0.0 Alpha

### DISCUSSION THREAD

**Discussion by wbooth** [mitre/caldera#2314] [General]
** Plugin UIs are still being updated so this will remain a pre-release until then

# New UI 
 
We are re-imagining the way end users interact with CALDERA. This includes large updates to the UI. 
Included is a new abilities screen to easily manage your extensive library. 

 
# API v2 
 
Calling all builders! For all those who build on the CALDERA platform we have a whole new API with full documentation. Currently docs are available once you start up the server. Look for a link at the bottom of the navigation menu "api docs" 

# C2 Channels 
 
We've introduced some new C2 channels, including: 
- Slack  
- SSH tunneling 
- FTP 
 
# Agent Updates 
- Sandcat agent support for new C2 channels (Slack, FTP, SSH tunneling) 
- New “proc” executor for Sandcat that will directly spawn processes using a provided executable path and arguments, rather than calling via PowerShell, sh, or cmd. 
- Sandcat agents can remove executors or update executor binary paths 
- Manx agents can properly run commands of longer durations. 
 
# Knowledge Service 
 
New service created to better manage facts and information during an operation or when performing analysis 
 
# File upload/download encoding 
Supports basic file encoding (plaintext and base64) for payload downloads and file uploads. To encode a downloaded payload or uploaded file, set the “x-file-encoding" HTTP header accordingly when making the download/upload request. Available data encoders are defined as Python modules in app/data_encoders. Currently supported encoders are “plain-text” and “base64” 
 
# Auth service 
Add support for custom login handlers, as well as a new SAML authentication plugin. 
 
# Other Changes 
 
- Dropped python 3.6 support and now testing for 3.7, 3.8, and 3.9
- We now support all browsers, Google Chrome is no longer the only supported browser

## New CALDERA Contributors
* @iguannalin made their first contribution in https://github.com/mitre/caldera/pull/2150
* @emmanvg made their first contribution in https://github.com/mitre/caldera/pull/2157
* @bleepbop made their first contribution in https://github.com/mitre/caldera/pull/2188
* @neptunia made their first contribution in https://github.com/mitre/caldera/pull/2224
* @Sloane4 made their first contribution in https://github.com/mitre/caldera/pull/2211
* @argaudreau made their first contribution in https://github.com/mitre/caldera/pull/2260

Thank you to the MANY builders of CALDERA on and off Github!

**Full Changelog**: https://github.com/mitre/caldera/compare/3.1.0...4.0.0-alpha

<hr /><em>This discussion was created from the release <a href='https://github.com/mitre/caldera/releases/tag/4.0.0-alpha'>4.0.0 Alpha</a>.</em>

**Comment by radioheado** [comment]
Hi Caldera team,
I'm really excited to try the new version! So far it looks so cool and convenient.
Here's one suggestion I just had while doing experiments on the new version.
The `training` section has the older version of the solution guide, where most of them work well. 
Long-time users would complete most of the tasks intuitively.
But still, I hope you'll update the solution guide accordingly so new users won't get confused.
For example, `Bootstrap abilities` guides us to copy the `ability_id` to the designated area, where now only takes textual description.

Anyways, thank you for the brilliant work!

**Reply by radioheado** [reply]
Here's another feedback: the scroll down function is not working for `payloads`. 
I have to click one of the six payloads presented initially to get my pointer in that area, then press the initial several times until the correct one is active.

**Reply by wbooth** [reply]
REALLY appreciate the kind words, everyone on the team works super hard. And good catch! we'll ensure that's fixed for the prod release
